apis/python/src/tiledbsoma/_dataframe.py

```diff
@@ -341,25 +334,28 @@ def read(
         Lifecycle:
             Experimental.
         """
-        del batch_size, platform_config  # Currently unused.
+        del batch_size  # Currently unused.
         _util.check_unpartitioned(partitions)
         self._check_open_read()
 
-        schema = self._handle.schema
-        query_condition = None
-        if value_filter is not None:
-            query_condition = QueryCondition(value_filter)
-
-        sr = self._soma_reader(
-            schema=schema,  # query_condition needs this
-            column_names=column_names,
-            query_condition=query_condition,
-            result_order=result_order,
+        ts = None
+        if self._handle._handle.timestamp is not None:
+            ts = (0, self._handle._handle.timestamp)
+
+        sr = clib.SOMADataFrame.open(
+            self._handle._handle.uri,
+            clib.OpenMode.read,
+            platform_config or {},
+            column_names or [],
+            _util.to_clib_result_order(result_order),
+            ts,
```

Comment:
one last bump on the suggestion to pass these as named arguments; particularly since we don’t have static typechecking on this, there is no obvious order for these